apps/fipsinstall.c

```diff
@@ -500,6 +557,16 @@ int fipsinstall_main(int argc, char **argv)
         if (!load_fips_prov_and_run_self_test(prov_name))
             goto end;
 
+        /* Warn if an attempt is being made to install in a non-approved manner */
+        if (compare_provider_version(3, 1, 0) >= 0 && self_test_onload == 0) {
+            BIO_printf(bio_err,
+                    "The FIPS provider must run self tests on load or the\n"
```

Comment:
This statement does not belong here like this.
We are talking about "requirements" that actually say what is being stated here.
The "best" we can do is direct the user to the "Security Policy" where they need to confirm their use is within the scope of the Security Policy. 

No definitive statements like this should exist.

And is this an error or not - it says it is a warning - but doesn't the goto end make it an error (haven't checked the context). It should not be an error.